mainwindow.cpp

```diff
@@ -74,12 +77,11 @@ void MainWindow::createActions() {
 
     connect(viewMemoryAction, &QAction::triggered, this->memoryViewerDialog,
             [this]() {
-        // TOOD(Moldo) change this when we implement the processor
-        unsigned char test[20] = {0x12, 0x10, 0x74, 0x44, 0x66, 0x43, 0x24, 0x76, 0x99, 0x56, 0x44, 0x22, 0x10, 0x12, 0x14, 0x24, 0x66};
-        memoryViewerDialog->setMemoryViewerData(reinterpret_cast<char*>(test));
+        memoryViewerDialog->setMemoryViewerData({reinterpret_cast<char*>(cpu->getMemory().data()), (int)cpu->getMemory().size()});
```

Comment:
Optionally since we need only the pointer to the start of the vector, I suggest  get memory returns an address, and create another function that returns the size of memory. Code will be clearer and more efficient.